Description: CVE_2014-3594: Fix XSS issue with the unordered_list filter
 When using the unordered_list filter in a Horizon table (as opposed to
 a template directly), autoescaping is not set by default and the input
 wasn't sanitised.
 .
Bug-Debian: https://bugs.debian.org/758930
Bug-Ubuntu: https://launchpad.net/bugs/1349491
From: 22edde980e0fb2238a62845bcd6922b39506b67b Mon Sep 17 00:00:00 2001
Author: Julie Pichon <jpichon@redhat.com>
Date: Tue, 29 Jul 2014 16:17:44 +0100
Change-Id: Id82eefe48ccb17a158751ec65d24f3ac779380ec
Last-Update: 2014-08-23
---
 openstack_dashboard/dashboards/admin/aggregates/tables.py | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

Index: horizon/openstack_dashboard/dashboards/admin/aggregates/tables.py
===================================================================
--- horizon.orig/openstack_dashboard/dashboards/admin/aggregates/tables.py	2014-08-23 07:50:20.000000000 +0800
+++ horizon/openstack_dashboard/dashboards/admin/aggregates/tables.py	2014-08-23 10:30:13.000000000 +0800
@@ -93,6 +93,10 @@
     return host_details
 
 
+def safe_unordered_list(value):
+    return filters.unordered_list(value, autoescape=True)
+
+
 class HostAggregatesTable(tables.DataTable):
     name = tables.Column('name', verbose_name=_('Name'))
     availability_zone = tables.Column('availability_zone',
@@ -100,11 +104,11 @@
     hosts = tables.Column(get_aggregate_hosts,
                           verbose_name=_("Hosts"),
                           wrap_list=True,
-                          filters=(filters.unordered_list,))
+                          filters=(safe_unordered_list,))
     metadata = tables.Column(get_metadata,
                              verbose_name=_("Metadata"),
                              wrap_list=True,
-                             filters=(filters.unordered_list,))
+                             filters=(safe_unordered_list,))
 
     class Meta:
         name = "host_aggregates"
@@ -123,7 +127,7 @@
     hosts = tables.Column(get_zone_hosts,
                           verbose_name=_('Hosts'),
                           wrap_list=True,
-                          filters=(filters.unordered_list,))
+                          filters=(safe_unordered_list,))
     available = tables.Column(get_available,
                               verbose_name=_('Available'),
                               status=True,
